students/perrymant/code-examples/cache/src/Cache.java

```diff
@@ -0,0 +1,167 @@
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * 2 methods:
+ * - Gives all users associated with 1 account
+ * - Find all the accounts associated with 1 user
+ */
+public class Cache {
+    // The List of users and accounts that are stored in the cache:
+    private List<User> users = new ArrayList<>();
```

Comment:
What's the time complexity right now?
I think we can make it O(1) if we create maps of user IDs to lists of accounts, (and the analogous case for accounts).
As a bonus, this will probably significantly simplify the code as well, once well-refactored.